### DISCUSSION TITLE
How to access pinia store elegant in quasar boot files

### DISCUSSION THREAD

**Discussion by maocis** [quasarframework/quasar#13704] [General - Components / Directives / etc] [answered]
Hello everyone
I am using Quasar CLI with vite，store with pinia
and I am apologize for my poor english
### Quasar info:
```
  quasar - 2.7.1 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.0.1 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.14.0 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.33 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.0.15
  pinia - 2.0.14 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 2.9.1 -- Native-ESM powered web dev build tool
  *None installed*
```
### my pinia store file:

```
import { defineStore } from 'pinia'
console.log('init pinia')
export const useCounterStore = defineStore('counter', {
    state: () => ({
        counter: 0,
    }),
    getters: {
        doubleCount: (state) => state.counter * 2,
    },
    actions: {
        increment() {
            this.counter++;
        },
    },
})
```

because boot file has 'store' property
[https://quasar.dev/quasar-cli-vite/boot-files#anatomy-of-a-boot-file](url)

if I use 'store' property like this:
```
import { useCounterStore } from 'stores/counter'
export default ({ app, router, store }) => {
    const use_store = useCounterStore()
    console.log('counter1', use_store.counter) // access counter by use_store
    console.log('counter2', store.state.value.counter.counter) // access counter with 'store' porperty
  // how to access pinia store here, for example access counter
}
```
I can access counter data correctly, now that I already import useCounterStore from store file, why I  need to 'store' porperty in boot file, so I changed boot file like this:
```
export default ({ app, router, store }) => {
    console.log('counter2', store.state.value.counter.counter) // access counter with 'store' porperty
  // how to access pinia store here, for example access counter
}
```
but I get error: 
`Cannot read properties of undefined (reading 'counter')`
although I removed useCounterStore, but I still get 'init pinia', dose this can prove store file run before boot file?
so, my question is: 
### How to access pinia store elegant in quasar boot files exactly？

**Comment by yusufkandemir** [comment, marked as answer]
When using Pinia, `store` is to pass to `useXYZStore` calls. When using stores outside Vue files(_e.g. in a boot file_) and doing SSR, you need to pass it. See: https://pinia.vuejs.org/ssr/#using-the-store-outside-of-setup

So, the way to utilize the `store` is like this:
```js
import { boot } from 'quasar/wrappers'
import { useCounterStore } from 'stores/counter'

export default boot(({ store }) => {
  const counterStore = useCounterStore(store)

  console.log(counterStore.counter)
})
```
It's optional when not doing SSR, so you can just ignore it if you are not doing SSR.

**Reply by maocis** [reply]
thanks for your reply
you mean no need to use 'store' property when not doing SSR?
Just use import?
```
import { boot } from 'quasar/wrappers'
import { useCounterStore } from 'stores/counter'

export default boot(({  }) => {
  const counterStore = useCounterStore()

  console.log(counterStore.counter)
})
```

**Reply by yusufkandemir** [reply]
> It's **optional** when not doing SSR, so **you can just ignore it** if you are not doing SSR.

Yes, the way you showed should work when not doing SSR.

**Reply by maocis** [reply]
yes, it's works well
thanks

**Comment by danbars** [comment]
I think this should be added to the docs.
Currently in the docs it is not clear:
1. How to use `store` property to access a specific pinia store defined using `defineStore`
2. That it is not needed to use it when not in SSR
